### ISSUE TITLE
Show result counts in the filters panel as results returned after all applied filters plus the new filter (react-search-refiners)

### ISSUE THREAD

**Issue by mgordin** [SharePoint/sp-dev-fx-webparts#641]
## Category

- [ ] Question
- [ ] Bug
- [x] Enhancement

## Authors

@FranckyC 
@wobba 

## Expected or Desired Behavior

It would be helpful if result counts in the filter panel reflected the number of results that would be returned after applying the new filter to the current results set, i.e., after any filters already there.

Currently, the result count shown is the number of results that would be seen if applying the filter to the original, unfiltered result set.  This is misleading for users, since they don't know how much adding sequential refiners actually refines their search results.  This also exacerbates the [bug I just posted](https://github.com/SharePoint/sp-dev-fx-webparts/issues/640#issue-367408364), since a user doesn't know when a combination of filters would produce 0 results (and thus totally break filtering).

**Comment by FranckyC** [comment]
@mgordin You're right. The best would be to follow the default SharePoint behavior. Scheduled for next release!

**Comment by FranckyC** [comment]
@mgordin Should be resovled. You can take a look.
